tests/src/test/scala/org/json4s/RichSerializerTest.scala

```diff
@@ -0,0 +1,50 @@
+package org.json4s
+
+import org.json4s.jackson.{JsonMethods, Serialization}
+import org.json4s.reflect.ScalaType
+import org.json4s.Extraction._
+import org.specs2.mutable.Specification
+
+class RichSerializerTest extends Specification {
+
+  object CustomTuple2Serializer extends RichSerializer[(_, _)] {
+
+    override def deserialize(implicit format: Formats): PartialFunction[(ScalaType, JValue), (_, _)] = {
+      case (scalaType, JArray(arr)) if classOf[(_, _)].isAssignableFrom(scalaType.erasure) =>
```

Comment:
The general motivation is to preserve deep type information when writing custom container classes. In particular, I've written custom tuple serializers because the `{"_1":...,"_2":...,...}` format didn't work for my use case (and they have bugs because of loss of type info).